hera_mc/cm_revisions.py

```diff
@@ -28,7 +28,11 @@ def get_revisions_of_type(rev_type, hpn, session=None):
     if rq == 'LAST':
         rq = get_last_revision(hpn, session)
     elif rq == 'ACTIVE':
-        rq = get_contemporary_revision(hpn, session)
+        if at_date is None:
+            print("Need a date to find active revision.")
```

Comment:
What's the outcome if `at_date is None`? I.e., what does a return value of `None` cause to happen? If the intention is that the program should abort, this should raise an exception instead of printing a message and returning None. If some backup behavior happens, the message should describe it; i.e., "Need a date to find the \"active\" revision; using the \"last\" revision instead", or something.

Stylistically, I like to structure these kinds of functions like this:

```python
def get_revisions_of_type(rev_type, hpn, at_date=None, session=None):
    rq = rev_type.upper()
    if rq == 'LAST':
        return get_last_revision(hpn, session)

    if rq == 'ACTIVE':
        if at_date is None:
            print("Need a date to find active revision.")
            return None
        else:
            return get_contemporary_revision(hpn, at_date, session)

    if rq == 'ALL':
        return get_all_revisions(hpn, session)

    return get_particular_revision(rq, hpn, session)
```